test/test_nn.py

```diff
@@ -7019,6 +7019,16 @@ def test_cross_entropy_loss_zero_div(self):
         input_2 = torch.rand([5, 0], dtype=torch.float32)
         torch.nn.CrossEntropyLoss()(input_1, input_2)
 
+    def test_cross_entropy_dim_parameter(self):
```

Comment:
We really want this within the cross-device tests to make sure we're covering CUDA as well ([link](https://github.com/pytorch/pytorch/blob/5210d7c4233f115d57a09cfe7d96c2a32a057f7c/test/test_nn.py#L11575) to location after the other device-specific cross entropy tests).

Reply:
Also, we need to make sure that the tests cover all three reduction types. In particular, I don't think `reduction='none'` will work in the current state.

You can follow what was done here to parametrize the test accordingly:
https://github.com/pytorch/pytorch/blob/5210d7c4233f115d57a09cfe7d96c2a32a057f7c/test/test_nn.py#L11555-L11556

Reply:
@jbschlosser, I've tried parameterizing the tests to cover the 3 reduction types, and tried to cover CUDA as well.